### PR TITLE
WT-5292 Test the garbage collection by checkpoint

### PR DESCRIPTION
**Pull request by kommiharibabu**
Perform the checkpoint garbage collection tests
by adding additional tests and also updating the existing
tests to remove the known limitations

### CODE REVIEW

File: test/suite/test_gc01.py [hunk 1]
```diff
@@ -69,7 +55,7 @@ def large_modifies(self, uri, value, ds, location, nbytes, nrows, commit_ts):
         session = self.session
         cursor = session.open_cursor(uri)
         session.begin_transaction()
-        for i in range(1, nrows):
+        for i in range(1, nrows + 1):
```
Comment (sueloverso) [inline test/suite/test_gc01.py]: This is an unusual usage so maybe a comment here about why we have to go to `nrows + 1`. Or alternatively, some other check could look for `nrows - 1` and this stays just `nrows`. I don't feel strongly though.

File: test/suite/test_gc01.py [hunk 2]
```diff
@@ -87,6 +73,26 @@ def check(self, check_value, uri, nrows, read_ts):
...
+
+        self.assertGreaterEqual(c[stat.conn.hs_gc_pages_visited][2], 0)
+        self.assertGreaterEqual(c[stat.conn.hs_gc_pages_removed][2], 0)
+        c.close()
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:', None, 'statistics=(all)')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
```
Comment (sueloverso) [inline test/suite/test_gc01.py]: I don't see this being called anywhere in the changes to this file. If it isn't used, it should be removed.

File: test/suite/test_gc01.py [hunk 3]
```diff
@@ -87,6 +73,26 @@ def check(self, check_value, uri, nrows, read_ts):
         session.rollback_transaction()
         self.assertEqual(count, nrows)
 
+    def check_gc_stats(self):
+        c = self.session.open_cursor(
+            'statistics:', None, 'statistics=(fast)')
```
Comment (sueloverso) [inline test/suite/test_gc01.py]: I think these lines can be merged into one? Does it matter that we specify `statistics=(fast)` here? Using `None` gets us the connection configuration which is `all` so anything we're looking for should be there and performance is not an issue.

File: test/suite/test_gc01.py (hunk 1 above)
Comment (tammybailey) [reply test/suite/test_gc01.py]: I changed the range to `(0, nrows)` everywhere it was used.

File: test/suite/test_gc01.py (hunk 2 above)
Comment (tammybailey) [reply test/suite/test_gc01.py]: I removed the function here and added it to the test files that were using it.

File: test/suite/test_gc01.py (hunk 3 above)
Comment (tammybailey) [reply test/suite/test_gc01.py]: No, it does not. Lines merged and code updated to `c = self.session.open_cursor('statistics:')`.

File: test/suite/test_gc01.py [hunk 4]
```diff
@@ -104,71 +110,77 @@ def test_gc(self):
...
 
-        # Check that the new updates are only seen after the update timestamp
-        #self.check(bigvalue, uri, nrows, 100)
+        # Check that the new updates are only seen after the update timestamp.
+        self.check(bigvalue2, uri, nrows, 100)
 
         # Load a slight modification with a later timestamp.
         self.large_modifies(uri, 'A', ds, 10, 1, nrows, 110)
         self.large_modifies(uri, 'B', ds, 20, 1, nrows, 120)
         self.large_modifies(uri, 'C', ds, 30, 1, nrows, 130)
```
[the remark this answers is not in the document]
Comment (tammybailey) [reply test/suite/test_gc01.py]: I added checks for all modifies across all tests that use them.

**Comment by ddanderson** [comment]
LGTM!

**Pull request merged**